lib/puppet/util/at_fork.rb

```diff
@@ -0,0 +1,35 @@
+require 'puppet'
+
+# A module for building AtFork handlers. These handlers are objects providing
+# pre/post fork callbacks modeled after those registered by the `pthread_atfork`
+# function.
+# Currently there are two AtFork handler implementations:
+# - a noop implementation used on all platforms except Solaris (and possibly
+#   even there as a fallback)
+# - a Solaris implementation which ensures the forked process runs in a different
+#   contract than the parent process. This is necessary for agent runs stared by
```

Comment:
This is the only comment I expect a change for. The others were just observations.